metrics/rpc_metrics.go

```diff
@@ -0,0 +1,40 @@
+package metrics
+
+import (
+	"time"
+
+	stdprometheus "github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+type Metrics interface {
+	RecordRPCMethodCall(method string, start time.Time)
+}
+
+type RPCMetrics struct {
```

Comment:
Huge nit but this type is applied against both eth json `RPC` methods and engine `API` methods.

Slightly annoying, but I think not worth a lot of consideration.

Reply:
Good point, I think it's probably worth renaming. I guess that all RPC methods are also technically [API methods](https://ethereum.org/en/developers/docs/apis/json-rpc/) , do you think that renaming this to `APIMetrics` would be more descriptive?

Reply:
Could do. Could leave as is. Up to you :)

Reply:
On second thought I'll opt to just leave it as is since the `engine` methods are also RPC calls as well